zookeeper-server/src/main/java/org/apache/zookeeper/server/backup/BackupUtil.java

```diff
@@ -151,14 +153,37 @@ public int compare(BackupFileInfo o1, BackupFileInfo o2) {
   /**
    * Creates a file name string for a backup file by appending a high zxid/long in Hex if it
    * doesn't already contain an ending zxid.
+   * It also adds the snapshot extension in case snapshot compression is enabled.
+   * E.g.) snapshot.123.snappy -> snapshot.123-456.snappy
    * @param standardName
    * @param highZxid
    * @return
    */
   public static String makeBackupName(String standardName, long highZxid) {
-    return standardName.indexOf('-') >= 0
-        ? standardName
-        : String.format("%s-%x", standardName, highZxid);
+    if (standardName.indexOf('-') >= 0) {
+      // standard name already contains the zxid interval endpoint and if snapshot compression
+      // is enabled, the standardName should already contain StreamMode extension, so just return
+      return standardName;
+    } else {
```

Comment:
NIT: we may not need this else clause. Same as following else when if will return something.